ReactorDebugAgent.init fails with the NoClassDefFoundError in 3.3.6.RELEASE

<!--- Provide a general summary of the issue in the Title above -->

The `ReactorDebugAgent.init()` has been broken in the latest release version `3.3.6.RELEASE` and it fails with the `NoClassDefFoundError`.

In `3.3.5.RELEASE` version everything works fine (just downgrade version back in a build.gradle.kts in the reproducer and `ReactorDebugAgent.init()` will work fine)

The issue was originally discovered in a project that uses __Spring-Boot__ after updating from `2.3.0.RELEASE` to `2.3.1.RELEASE` (that bumps the __reactor__ version from `3.3.5.RELEASE` to `3.3.6.RELEASE`). 
But I was able to make a small reproducer attached below.

<!--- /!\ Make sure to follow the Contribution Guidelines, notably for security issues and questions:
https://github.com/reactor/.github/blob/master/CONTRIBUTING.md
https://pivotal.io/security
https://github.com/reactor/.github/blob/master/CONTRIBUTING.md#question-do-you-have-a-question
-->

## Expected Behavior

`ReactorDebugAgent.init()` completes successfully

## Actual Behavior
<!--- Tell us what happens instead of the expected behavior. -->
`ReactorDebugAgent.init()` fails with the `NoClassDefFoundError`:
<details>
  <summary>Stacktrace</summary>

```
Exception in thread "main" java.lang.NoClassDefFoundError: net/bytebuddy/agent/ByteBuddyAgent
	at reactor.tools.agent.ReactorDebugAgent.init(ReactorDebugAgent.java:56)
	at Application.main(Application.java:5)

Caused by: java.lang.ClassNotFoundException: net.bytebuddy.agent.ByteBuddyAgent
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:602)
Caused by: java.lang.ClassNotFoundException: net.bytebuddy.agent.ByteBuddyAgent

	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 2 more
```
  
</details>

## Steps to Reproduce
<!---Provide a link to a live example, or an unambiguous set of steps to
reproduce this bug, eg. a unit test. Include code to reproduce, if relevant. -->

1. A Gradle project of the following structure:
<details>
  <summary>gradle.build.kts</summary>

```gradle
plugins {
    java
    application
}

repositories {
    jcenter()
}

dependencies {
    // 3.3.5.RELEASE works fine
    implementation("io.projectreactor:reactor-tools:3.3.6.RELEASE")
}

application {
    mainClassName = "Application"
}

```

</details>

<details>
  <summary>src/main/java/Application.java</summary>

```java
import reactor.tools.agent.ReactorDebugAgent;

public class Application {
    public static void main(String[] args) {
        ReactorDebugAgent.init();
    }
}

```

</details>

2. Run `./gradlew run`


## Possible Solution
<!--- Not obligatory, but you can suggest a fix/reason for the bug. -->

I've seen some commits in the last [release](https://github.com/reactor/reactor-core/compare/v3.3.6.RELEASE...master) that manipulate with version format, so that can be the reason for the bug.

I'll try to investigate the reason in a more detailed way, but at the moment I have no idea what specifically led to the bug

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in. -->
<!--- Especially, always include the version(s) of Reactor library/libraries you used! -->

* Reactor version(s) used: 3.3.6
* JVM version: openjdk version "14" 2020-03-17
OpenJDK Runtime Environment (build 14+36-1461)
OpenJDK 64-Bit Server VM (build 14+36-1461, mixed mode, sharing)
* Gradle version: 6.5
* OS and version: macOS 10.15.4

--- a/gradle/setup.gradle
+++ b/gradle/setup.gradle
@@ -37,6 +37,12 @@ task javadocJar(type: Jar) {
 }
 
 publishing {
+    repositories {
+        maven {
+           name = "mock"
+           url = "${rootProject.buildDir}/repo"
+        }
+    }
 	publications {
 		mavenJava(MavenPublication) {
 			from components.java

--- a/reactor-tools/build.gradle
+++ b/reactor-tools/build.gradle
@@ -186,5 +186,14 @@ tasks.withType(Test).all {
     useJUnitPlatform()
 }
 
-//add jar and original jar to the publication
-publishing.publications.mavenJava.artifact(shadowJar)
+// make the shadow jar the default artifact
+['apiElements', 'runtimeElements'].each {
+    configurations.getByName(it).outgoing {
+        artifacts.clear()
+        artifact(shadowJar)
+    }
+}
+configurations.runtime.outgoing.artifacts.clear()
+
+//add original jar to the publication
+publishing.publications.mavenJava.artifact(jar)

